go/vt/vtgate/mysql_protocol_test.go

```diff
@@ -92,6 +93,35 @@ func TestMySQLProtocolStreamExecute(t *testing.T) {
 	}
 }
 
+func TestMySQLProtocolExecuteUseStatement(t *testing.T) {
+	createSandbox(KsTestUnsharded)
+	hcVTGateTest.Reset()
+	hcVTGateTest.AddTestTablet("aa", "1.1.1.1", 1001, KsTestUnsharded, "0", topodatapb.TabletType_MASTER, true, 1, nil)
+
+	c, err := mysqlConnect(&mysql.ConnParams{DbName:"@master"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer c.Close()
+
+	qr, err := c.ExecuteFetch("show vitess_target", 1, false)
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, "VARCHAR(\"@master\")", qr.Rows[0][0].String())
```

Comment:
For completeness I think we should have tests that show the expected behavior and not just the results of `show vitess_target`. Specifically, when in `@master` mode, run a simple `select * from table`  query against tables that are in more than one keyspace to show that the multi keyspace routing works.

Then when we force it to be using `TestUnsharded` show that the above no longer works for the tables in other keyspaces. 

We should also prove the behavior is correct when (a) the initial ConnParams contains no DbName and (b) that when ConnParams contains `TestUnsharded` you can still override to another keyspace or to `@master` to get the multi-keyspace mode.

(a) a test case that things work as expected when we connect without an intial

Reply:
I couldn't figure out how to do what you asked for without reams of test set up code but I think I've added sufficient tests to prove that we're using the vitess target for routing. I query on the `@replica` target with only `master` tablets and it fails with the expected error message.